libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -398,6 +398,13 @@ void ContractLevelChecker::checkExternalTypeClashes(ContractDefinition const& _c
 					);
 			}
 		for (VariableDeclaration const* v: contract->stateVariables())
+		{
+			if (v->isConstant() && v->type()->containsNestedMapping())
```

Comment:
Is it possible to move this to type checker? Or is too late?

Reply:
Somewhere in here:
https://github.com/ethereum/solidity/blob/a83ad58109604bb4795a246b2ac4844faa2831af/libsolidity/analysis/TypeChecker.cpp#L499

Reply:
Unfortunately it seems to be too late indeed. The assertion happens within `ContractLevelChecker`. Here is my stack:

```
solc.exe!solidity::frontend::StructType::nativeMembers(const solidity::frontend::ASTNode * __formal) Line 2406	C++
solc.exe!solidity::frontend::Type::members(const solidity::frontend::ASTNode * _currentScope) Line 355	C++
solc.exe!solidity::frontend::FunctionType::FunctionType(const solidity::frontend::VariableDeclaration & _varDecl) Line 2852	C++
[Inline Frame] solc.exe!std::make_unique(const solidity::frontend::VariableDeclaration &) Line 2539	C++
[Inline Frame] solc.exe!solidity::frontend::TypeProvider::createAndGet(const solidity::frontend::VariableDeclaration &) Line 203	C++
solc.exe!solidity::frontend::TypeProvider::function(const solidity::frontend::VariableDeclaration & _varDecl) Line 426	C++
solc.exe!solidity::frontend::ContractLevelChecker::checkExternalTypeClashes(const solidity::frontend::ContractDefinition & _contract) Line 403	C++
solc.exe!solidity::frontend::ContractLevelChecker::check(const solidity::frontend::ContractDefinition & _contract) Line 92	C++
solc.exe!solidity::frontend::ContractLevelChecker::check(const solidity::frontend::SourceUnit & _sourceUnit) Line 77	C++
solc.exe!solidity::frontend::CompilerStack::analyze() Line 374	C++
solc.exe!solidity::frontend::CompilerStack::compile(solidity::frontend::CompilerStack::State _stopAfter) Line 517	C++
solc.exe!solidity::frontend::CommandLineInterface::processInput() Line 1487	C++
solc.exe!main(int argc, char * * argv) Line 60	C++
```

Reply:
Can you add an assert for `v->type()` before this? Not entirely sure if all type resolution is complete at this stage.

Reply:
Perhaps we can skip going into the if condition on line 408 if `v->isConstant` is not a value type? Or maybe if it contains a nested mapping? Then the type checker can handle the error properly. Sounds a bit hacky, so not sure.